restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCConnection.java

```diff
@@ -560,10 +563,19 @@ void initializeWebrtc(boolean videoEnabled)
         for (String permission : MANDATORY_PERMISSIONS) {
             if (context.checkCallingOrSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                 logAndToast("Permission " + permission + " is not granted");
-                // TODO: return error to RCConnection listener
+                listener.onPermissionNotGranted(this, permission);
```

Comment:
@mehturt sorry for being late but I've been very busy lately. So if you make a new call in (3), a new RCConnection object will be created which will have it's state re-initialized, so it will no longer be BUSY. It will start again from scratch. Is this the scenario you are describing?